invenio_oauthclient/utils.py

```diff
@@ -160,27 +160,20 @@ def oauth_unlink_external_id(external_id):
                                      method=external_id['method']).delete()
 
 
-def is_local_url(target):
-    """Determine if URL is a local.
-
-    :param target: The URL to check.
-    :returns: ``True`` if the target is a local url.
-    """
-    server_name = current_app.config['SERVER_NAME']
-    test_url = urisplit(target)
-    return not test_url.host or test_url.scheme in ('http', 'https') and \
-        server_name == test_url.host
-
-
 def get_safe_redirect_target(arg='next'):
     """Get URL to redirect to and ensure that it is local.
 
     :param arg: URL argument.
     :returns: The redirect target or ``None``.
     """
     for target in request.args.get(arg), request.referrer:
-        if target and is_local_url(target):
-            return target
+        if target:
+            target = urisplit(target)
+            if target.path:
+                allowed_hosts = current_app.config.get('APP_ALLOWED_HOSTS', [])
+                if target.host in allowed_hosts:
```

Comment:
So, this is the safe and dumb-proof way, but might come in the way of people from testing/debugging things... @lnielsen @egabancho WDYT?

Reply:
getting in their way in what sense?